### PR TITLE
Add new error for empty Islands

### PR DESCRIPTION
**Pull request by bryancroteau-MSFT**
Empty islands such as $"{}" are an error in C# and should be an error in PowerFx.

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/StringInterpolate.txt [hunk 1]
```diff
@@ -109,4 +109,4 @@ Errors: Error 6-15: Invalid argument type (Record). Expecting a Text value inste
 Errors: Error 9-10: Unexpected characters. Characters are used in the formula in an unexpected way.|Error 10-10: Unexpected characters. Characters are used in the formula in an unexpected way.
 
 >> $"Hello {}"
-"Hello "
+Errors: Error 9-10: Empty expressions cannot appear inside an interpolated string.
```
Comment (lesaltzm) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/StringInterpolate.txt:112]: also a test of empty string + empty island,`$"{}"`

File: src/strings/en-US/PowerFxResources.resw [hunk 2]
```diff
@@ -6361,6 +6361,14 @@
     <value>Check the types of the expressions inside the interpolated string.</value>
     <comment>How to fix message for an error</comment>
   </data>
+  <data name="ErrorResource_ErrEmptyIsland_ShortMessage" xml:space="preserve">
+    <value>Empty expressions cannot appear inside an interpolated string.</value>
+    <comment>Error message.</comment>
```
Comment (lesaltzm) [inline src/strings/en-US/PowerFxResources.resw:6366]: @CarlosFigueiraMSFT , for localization teams, do we need to provide some context on "interpolated"? I know it's used in a few other messages here, but maybe we should comment that it's translated like C# docs do?

Comment (CarlosFigueiraMSFT) [reply src/strings/en-US/PowerFxResources.resw:6366]: It never hurts to provide additional context :-)

Comment (bryancroteau-MSFT) [reply src/strings/en-US/PowerFxResources.resw]: Added some context

**Review by MikeStall — APPROVED** [review]
:shipit:

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

**Review by lesaltzm — APPROVED** [review]
